XMonad/Layout/IndependentScreens.hs

```diff
@@ -122,20 +148,17 @@ onCurrentScreen f vws = screen . current >>= f . flip marshall vws
 countScreens :: (MonadIO m, Integral i) => m i
 countScreens = fmap genericLength . liftIO $ openDisplay "" >>= liftA2 (<*) getScreenInfo closeDisplay
 
--- | This turns a naive pretty-printer into one that is aware of the
--- independent screens. That is, you can write your pretty printer to behave
--- the way you want on virtual workspaces; this function will convert that
--- pretty-printer into one that first filters out physical workspaces on other
--- screens, then converts all the physical workspaces on this screen to their
--- virtual names.
+-- | This turns a pretty-printer into one that is aware of the
+-- independent screens. The converted pretty-printer will first 
+-- filters out physical workspaces on other screens, then sort 
+-- all the physical workspaces on this screen.
```

Comment:
We shouldn't drop the mention of virtual names, I think. They're not involved in sorting any more, but marshallPP still converts the names in ppRename.

How about this:

> This turns a pretty-printer into one that is aware of the independent screens. The converted pretty-printer first filters out physical workspaces on other screens, then converts all the physical workspaces on this screen to their virtual names.
> Note that 'ppSort' still operates on physical (marshalled) workspace names, otherwise functions from "XMonad.Util.WorkspaceCompare" wouldn't work. If you need to sort on virtual names, see 'marshallSort'.

Reply:
Thanks very much. I think your version is clearer than mime. I forgot to mention about 'ppSort' here.